Scripts/modelsystem.py

```diff
@@ -156,11 +156,11 @@ def _add_internal_demand(self, previous_iter_impedance, is_last_iteration):
                         purpose, "car_leisure", purpose_impedance)
             else:
                 demand = purpose.calc_demand()
-                if is_last_iteration:
-                    purpose.print_data()
                 if purpose.dest != "source":
                     for mode in demand:
                         self.dtm.add_demand(demand[mode])
+        if is_last_iteration:
+            self._export_resultdata()
```

Comment:
I think we should move this to the end of `run_iteration()`. Then it will automatically affect agent model as well.